upstream.go

```diff
@@ -1344,3 +1395,47 @@ func (uc *upstreamConn) SendMessageLabeled(downstreamID uint64, msg *irc.Message
 	}
 	uc.SendMessage(msg)
 }
+
+// TODO: handle moving logs when a network name changes, when support for this is added
+func (uc *upstreamConn) Logfn(entity string, format string, a ...interface{}) {
```

Comment:
Maybe we could rename this to `AppendLog` or `WriteLog`? The `fn` suffix doesn't add a lot of value.

Reply:
OK,  `AppendLog` is fine